app/src/main/java/com/android/voyageur/model/trip/TripRepositoryFirebase.kt

```diff
@@ -64,6 +64,34 @@ class TripRepositoryFirebase(private val db: FirebaseFirestore) : TripRepository
         }
   }
 
+  /**
+   * Fetches trips that are discoverable and not created by the user.
+   *
+   * @param userId The ID of the user.
+   * @param onSuccess The callback to be invoked when the trips are fetched successfully.
+   * @param onFailure The callback to be invoked when an error occurs.
+   */
+  override fun getFeed(
+      userId: String,
+      onSuccess: (List<Trip>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection(collectionPath)
+        .whereEqualTo("discoverable", true)
+        .get()
+        .addOnSuccessListener { result ->
+          val trips =
+              result
+                  .map { document -> document.toObject(Trip::class.java) }
+                  .filter { it.creator != userId && !it.participants.contains(userId) }
```

Comment:
We should ensure that it.participants is non-null before calling .contains(userId). If participants is null, it could throw a NullPointerException.

Reply:
participants will never be null, even if it is empty it will be a valid (but empty) list which supports this operation.

Reply:
Ok, if the field can never be null then leave it, I have just seen a warning there when checking out your branch in my IDE.